packages/ckeditor5-image/src/image/imageediting.js

```diff
@@ -114,6 +135,8 @@ export default class ImageEditing extends Plugin {
 			.add( viewFigureToModel() );
 
 		editor.commands.add( 'imageInsert', new ImageInsertCommand( editor ) );
+		editor.commands.add( 'imageBlockToInline', new ImageBlockToInlineCommand( this.editor ) );
+		editor.commands.add( 'imageInlineToBlock', new ImageInlineToBlockCommand( this.editor ) );
```

Comment:
Why do we need 2 separate commands for this? 

This is a simple state machine and `ImageTypeToggleCommand()` should do the trick. I checked implementations and they are fairly simple. The unified command:

*   would be enabled if the selected model element is either `image` or `imageInline`,
*   would have `value` indicating which one is it (e.g. for the UI to bind to), either `'image'` or `'imageInline'`
*   would have `execute( options )` that toggles the kind of image
    *   (in the future) could support `{String} options.type` that allows integrators to enforce `image` or `imageInline` regardless of the current model state